torchaudio/datasets/speechcommands.py

```diff
@@ -79,9 +90,30 @@ def __init__(self,
                     download_url(url, root, hash_value=checksum, hash_type="md5")
                 extract_archive(archive, self._path)
 
-        walker = walk_files(self._path, suffix=".wav", prefix=True)
-        walker = filter(lambda w: HASH_DIVIDER in w and EXCEPT_FOLDER not in w, walker)
-        self._walker = list(walker)
+        def _load_list(*filenames):
+            output = []
+            for filename in filenames:
+                filepath = os.path.join(self._path, filename)
+                with open(filepath) as fileobj:
+                    output += [os.path.normpath(os.path.join(self._path, line.strip())) for line in fileobj]
+            return output
+
+        if subset == "validation":
+            self._walker = _load_list("validation_list.txt")
+        elif subset == "testing":
+            self._walker = _load_list("testing_list.txt")
+        elif subset == "training":
```

Comment:
I see there is a behavior inconsistency between `"training"/None` and `"validation"/"testing"`.

If certain valid files are removed from the dataset, then this dataset implementation will keep working for `"training"/None` case with less files, but for `"validation"/"testing"` it will raise an exception.

What's your take on that?

Reply:
"testing", "validation", "training" are defined by the dataset by the training/validation files. It is technically undefined by the dataset outside of that. Given how the three are defined by the dataset in those files, as a user, I'd expect changes to those file to propagate. I'd say it'd be fair to add a quick note in the docstring.

Reply:
Added.

Reply:
> It is technically undefined by the dataset outside of that.

I agree with that, however IIRC, one of the goal of torchaudio's dataset implementation is to make it easy to modify the dataset. Something along the line of point 3&4 of https://github.com/pytorch/audio/issues/852#issuecomment-670590773 . With this rule, we have to take the extra step to think through what kind of modification is valid/invalid and what is the expected behavior, and put it in the implementation.

I am trying to raise the awareness of it.

Reply:
And of course, if we are just getting rid of the easy modification of the dataset, we do no need think about the any modification to dataset.

Reply:
Actually, SpeechCommands does explain how those two files are generated, and how to generalize their approach, see [README](https://github.com/sankalp2610/Speech_Command_Recognition).